waftools/checks/custom.py

```diff
@@ -67,13 +68,20 @@ def check_lua(ctx, dependency_identifier):
         ( '51obsd', 'lua51 >= 5.1.0'), # OpenBSD
         ( '51deb',  'lua5.1 >= 5.1.0'), # debian
         ( '51fbsd', 'lua-5.1 >= 5.1.0'), # FreeBSD
-    ]
+    ])
 
     if ctx.options.LUA_VER:
-        lua_versions = \
-            [lv for lv in lua_versions if lv[0] == ctx.options.LUA_VER]
+        lua_versions = OrderedDict(
+            [(k, v) for k, v in lua_versions.items() if k == ctx.options.LUA_VER]
+        )
 
-    for lua_version, pkgconfig_query in lua_versions:
+    # luajit < 2.1.0 is broken on macOS Catalina, issues #7512
```

Comment:
We should never do this. If it's broken on Catalina, then users should not install it on Catalina, or the package manager should disable it on Catalina.